### PR TITLE
Fix ipv6 parsing in client

### PR DESCRIPTION
**Pull request by miguelxpn**
Old implementation of function ParseConnectionString would parse ipv6 incorrectly

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks for this.

What would be great is to see this code lifted out as a distinct function
and then some unit tests added.

**Comment by otoolep** [comment]
Scrap that. It's already a function. What I mean is can you add a unit test showing what would fail to parse correctly previously?

**Comment by miguelxpn** [comment]
I added a test with ipv6 that fails with the previous implementation and works with the new one. No idea why the other test failed though, did I do something wrong when adding the new test?

**Comment by miguelxpn** [comment]
Hmmm, weird, sent it again and now everything passed, seems there's an intermittent problem that's caused some tests to fail in the meta package.

**Comment by otoolep** [comment]
Yeah, known issue with the tests in that package.

File: client/influxdb_test.go [hunk 1]
```diff
@@ -547,3 +547,14 @@ func TestClient_NoTimeout(t *testing.T) {
 		t.Fatalf("unexpected error.  expected %v, actual %v", nil, err)
 	}
 }
+
+func TestClient_ParseConnectionStringi_Ipv6(t *testing.T) {
```
Comment (otoolep) [inline client/influxdb_test.go]: Is that a rogue 'i' in the test name?

Comment (otoolep) [inline client/influxdb_test.go]: Also, Go style dictates that this test should be named `TestClient_ParseConnectionString_IPv6`.

**Comment by otoolep** [comment]
@corylanou

**Comment by corylanou** [comment]
+1

**Comment by miguelxpn** [comment]
@otoolep there was indeed a rogue 'i' in there! Fixed the test name, thanks for the feedback!

**Comment by otoolep** [comment]
Ref: https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers

**Pull request merged**